numba/cuda/simulator/kernel.py

```diff
@@ -86,11 +86,16 @@ def fake_arg(arg):
                 )
 
                 if isinstance(arg, np.ndarray) and arg.ndim > 0:
-                    return wrap_arg(arg).to_device(retr)
+                    ret = wrap_arg(arg).to_device(retr)
                 elif isinstance(arg, ArgHint):
-                    return arg.to_device(retr)
+                    ret = arg.to_device(retr)
+                elif isinstance(arg, np.void):
+                    ret = FakeCUDAArray(arg)  # In case a np record comes in.
                 else:
-                    return arg
+                    ret = arg
+                if isinstance(ret,FakeCUDAArray):
+                    return FakeWithinKernelCUDAArray(ret)
```

Comment:
This is where FakeWithinKernelCUDAArrays get made - when you ship arguments to a kernel.

Reply:
Got it, I somehow glossed over this multiple times - thanks!